File: 프로그래머스/lv0/120893. 대문자와 소문자/대문자와 소문자.js
function solution(my_string) {
    let answer = '';
    for(let i=0; i<my_string.length; i++) {
        if(my_string[i] === my_string[i].toUpperCase()) {
            answer += my_string[i].toLowerCase();
        }else{
            answer += my_string[i].toUpperCase();
        }
    }
    return answer;
}
/*
    접근법 :
    들어온 문자열을 대문자는 소문자로, 소문자는 대문자로 바꿔야 한다.
    빈 문자열을 만들어 놓고 들어온 문자열을 순회하면 대,소문자로 바꾸어 넣는 방법을 해보자
    answer -> 대,소문자가 바뀐 문자가 들어갈 변수
    for문을 실행해 조건문으로 대소문자인지 판별후, 그에 맞게 넣어준다.

    다른 접근법 :
    
    1. 들어온 문자열을 문자 하나,하나 배열로 만들어 result에 담아줌
       result 기준으로 순회하며 풀어나가는 방법
    function solution(my_string) {
        let result = my_string.split('');
        for(let i=0; i<result.length; i++) {
            if(result[i] === result[i].toUpperCase()) {
                result[i] = result[i].toLowerCase();
            }else{
                result[i] = result[i].toUpperCase();
            }
        }
        return result.join('');
    }

    2. 유니코드를 이용한 방법
       'A' ~ = 65 ~, 'a' ~ = 97 ~
       즉, 97보다 작으면 대문자
    function solution(my_string) {
        let result = my_string.split('');
        for(let i=0; i<result.length; i++) {
            if(result[i].charCodeAt() < 97) {
                result[i] = result[i].toLowerCase();
            }else{
                result[i] = result[i].toUpperCase();
            }
        }
        return result.join('');
    }


*/

File: 프로그래머스/lv0/120893. 대문자와 소문자/대문자와 소문자.java
class Solution {
    public String solution(String my_string) {
        String answer = "";
        for(int i=0; i<my_string.length(); i++) {
            if(true == Character.isUpperCase(my_string.charAt(i))) {
                answer += Character.toLowerCase(my_string.charAt(i));
            }else {
                answer += Character.toUpperCase(my_string.charAt(i));
            }
        }
        return answer;
    }
}
/*
    접근법 :
    대소문자로 바뀐 문자들이 들어올 문자열 선언 -> answer
    들어온 문자열을 순회하며 문자들을 뽑아온다 -> .charAt();
    뽑아온 문자가 대문자인지, 소문자인지 판별 -> isUpper(Lower)Case();
    조건 대로 빈 문자열에 넣어준 다음 반환해준다.

    
*/